### ISSUE TITLE
Local captainhook.json support

### ISSUE THREAD

**Issue by boesing** [CaptainHookPhp/captainhook#83]
# Feature Request

Hey there,

We are many developers and thus, some of us have a `docker` configuration, others may have `vagrant` and so on.

I would prefer having a way to create example `local` configurations, where our developers can re-configure the projects configuration to their environment.

### Project configuration
*captainhook.json*
```json
{
	"config": {
		"verbosity": "normal"
	}
}
```

### Developers own configuration which is excluded from VCS
*captainhook.local.json*
```json
{
	"config": {
		"verbosity": "verbose",
		"run-mode": "docker",
		"run-exec": "docker exec whatever"
	}
}
```

The deal could be, that the `local` file is *always* next to the `captainhook.json`.
This might make the implementation quite easy.

What do you think about it?

**Comment by giuseppemorelli** [comment]
+1 for the issue

and add this problem with a team that shares docker and pc/mac with software installed on host.
If file captainhook.json is committed in git repository it's a problem to edit.
Ignore the file on git is a problem if someone makes some update

```json
{
    "config": {
         "run-mode": "docker",
    }
}
```

The best option is to create the configuration file like dotenv component of symfony

https://symfony.com/doc/current/components/dotenv.html

**Comment by sebastianfeldmann** [comment]
The Cap'n supports config file inclusion, does that help?

You could commit a `captainhook.inc.json` with all your hooks and add files like `captainhook.json.sample.docker` `captainhook.json.sample.vagrant`  with something like this.

```
{
  "config": {
    "includes": [
      "captainhook.inc.json"
    ]
  }
}
```

Developers would have to choose and copy the right config file to `captainhook.json` and then everything would work.

The other option I see is something like the `local` configuration file you proposed. But I think I would prefer something like `captainhook.config.json` that is only allowed to overwrite the Cap'n `config` settings and is not allowed to add any custom hooks.

What do you think about using environment variables to overwrite config settings?

**Comment by boesing** [comment]
What would happen if the file is not available, which would be the default as it is a per user config and thus excluded from VCS.

And yes, the local config should be only allowed to override config. It should not be able to provide additional hooks.

**Comment by sebastianfeldmann** [comment]
Fixed in version `5.3.0`

For details have a look at the [documentation](https://captainhookphp.github.io/captainhook/configure.html#settings-overwrite )
